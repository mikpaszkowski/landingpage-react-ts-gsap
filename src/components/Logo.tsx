import * as React from "react";
import { ReactComponent as Ellipse2 } from "../assets/images/svg-images/ellipse-2.svg";
import { ReactComponent as Ellipse1 } from "../assets/images/svg-images/ellipse-1.svg";
import styled from "styled-components";
import { colors } from "../styles/styleConstants";

const LogoWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  position: relative;
  width: 17.3rem;
  height: 4.7rem;
`;

const LogoText = styled.p`
  font-size: 4rem;
  font-style: normal;
  font-weight: normal;
  text-align: center;
  width: 10.2rem;
  line-height: 117%;
  color: ${(props: LogoProps) =>
    props.inverse ? "white" : colors.primaryFont};
`;

const Ellipse1Absolute = styled(Ellipse1)`
  position: absolute;
  top: 0.7rem;
  right: 0.3rem;
`;

interface LogoProps {
  inverse?: boolean | null;
}

export const Logo = ({ inverse }: LogoProps) => {
  return (
    <LogoWrapper>
      <Ellipse1Absolute />
      <Ellipse2 />
      <LogoText inverse={inverse}>Medli</LogoText>
    </LogoWrapper>
  );
};
