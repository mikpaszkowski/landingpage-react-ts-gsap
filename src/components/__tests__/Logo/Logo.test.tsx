import { cleanup } from "@testing-library/react";
import "jest-styled-components";
import { renderWithLightTheme, mountWithLightTheme } from "../helpers/index";
import { Logo } from "../../Logo/Logo";

afterEach(() => {
  cleanup();
});

test("check render correctly <Logo /> component", () => {
  const tree = renderWithLightTheme(<Logo />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe("check Logo component - lightTheme", () => {
  const wrapper = mountWithLightTheme(<Logo />).find(Logo);

  test("should have p tag element - Medli", () => {
    expect(wrapper.find("p").text()).toEqual("Medli");
  });

  test("should have two svg tag element", () => {
    expect(wrapper.find("svg").length).toEqual(2);
  });

  test("should have first svg image: ellipse-1.svg", () => {
    expect(wrapper.find("svg").at(0).text()).toEqual("ellipse-1.svg");
  });

  test("should have second svg image: ellipse-1.svg", () => {
    expect(wrapper.find("svg").at(1).text()).toEqual("ellipse-2.svg");
  });
});
