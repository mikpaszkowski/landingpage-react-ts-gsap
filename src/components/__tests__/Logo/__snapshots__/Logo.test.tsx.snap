// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check render correctly <Logo /> component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  width: 17.3rem;
  height: 4.7rem;
  cursor: pointer;
}

.c2 {
  font-size: 4rem;
  font-style: normal;
  font-weight: normal;
  text-align: center;
  width: 10.2rem;
  line-height: 117%;
  color: #131E07;
}

.c1 {
  position: absolute;
  top: 0.7rem;
  right: 0.3rem;
}

<div
  className="c0"
>
  <svg
    className="c1"
  >
    ellipse-1.svg
  </svg>
  <svg>
    ellipse-2.svg
  </svg>
  <p
    className="c2"
  >
    Medli
  </p>
</div>
`;
